baybe/simulation/core.py

```diff
@@ -156,6 +162,12 @@ def simulate_experiment(
             )
             break
 
+        # Temporary workaround to enable returning incomplete simulations
+        except Exception as ex:
+            if _return_on_error:
```

Comment:
Two things:
* because we have the upcoming refactoring, I consider this PR to merely unblock people (like @brandon-holt) but by no means introduce any breaking changes, which the other approach would do (i.e. the function would behave differently than before)
* Just returning seems not enough to me (see [here](https://github.com/emdgroup/baybe/issues/212#issuecomment-2065956660)) because I would want not only access to the data, but easily identify that something went wrong + get the corresponding error message etc. Especially since these simulations are often long running jobs...

Haven't fully fleshed out the details yet, but I could imagine something like having a `SimulationResult` method and a `SimulationResults` method (or similar) that allow you to access everything. I.e. `SimulationResult` would have a field containing the actual results, but also a field with an error code, a field with the raise exception etc.. Perhaps even meta-information like duration of the simulation etc, could be quite useful. Then `SimulationResults` could contain a list of the individual results together with appropriate helpers like `.collect` etc...

From the user perspective the only noticeable change would be the additional `collect` call, but you would have clear overview of what went wrong and how you want to handle it (e.g. `collect` could have flags like `drop_incomplete` etc)

Reply:
closing this thread in order to have a single point of discussion